lib/Transport.js

```diff
@@ -39,6 +39,7 @@ Transport._defaultError = function(response) {
 Transport.prototype._action = function(method, url, data, callback) {
   // console.log(method, url, JSON.stringify(data, null, 2));
   var someRequest = request[method](url);
+  someRequest.set("Content-Type", "application/vnd.api+json");
   this._attachAuthToRequest(someRequest);
   if (method === "get") {
```

Comment:
The matching `else` block on line 46 just outside of github's comment-range, if it becomes `} else if (data) {` to validate a payload exists, we'll be sending requests that:
1. Claim to be JSON
2. Are always valid JSON
...and the test suite should pass all the core tests. There may be some new CORS errors judging by a local test run, but they're nothing for you to worry about :)